app/models/notification.rb

```diff
@@ -0,0 +1,21 @@
+class Notification < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+  belongs_to :user
+  belongs_to :course
+
+  scope :center_popup, -> { where(type: 'CenterPopup') }
+  scope :right_side_popup, -> { where(type: 'RightSidePopup') }
+  scope :email, -> { where(type: 'EmailNotification') }
+
+  def self.types
```

Comment:
hmm... this may be unnecessary. Type would be the class name which inherits Notification class. Since I did not define create_notification method in Notification class, nobody can make type beyond those 3 kinds of types.